test/integration/concrete/lockup/withdraw-hooks/withdrawHooks.tree

```diff
@@ -0,0 +1,21 @@
+withdrawHooks.t.sol
+├── given the recipient is different than the sender
+│   ├── when the caller is unknown
+│   │   ├── it should make one hook call to the sender
+│   │   └── it should make one hook call to the recipient
+│   ├── when the caller is an approved third party
+│   │   ├── it should make one hook call to the sender
+│   │   └── it should make one hook call to the recipient
+│   ├── when the caller is the sender
+│   │   ├── it should not make any hook call to the sender
+│   │   └── it should make one hook call to the recipient
+│   └── when the caller is the recipient
+│       ├── it should make one hook call to the sender
+│       └── it should not make any hook call to the recipient
+└── given the recipient is same as the sender
+    ├── when the caller is unknown
+    │   └── it should make one hook call to the sender
+    ├── when the caller is an approved third party
+    │   └── it should make one hook call to the sender
+    └── when the caller is the sender
+        └── it should not make any hook call to the sender
```

Comment:
We typically test the less complicated branches first, but I don't think it's worth refactoring the code just for this nitpick of mine now.

Just sharing this feedback FYI, @smol-ninja.

Reply:
Sure. Thanks for pointing it out.